src/ctia/bulk/core.clj

```diff
@@ -99,16 +93,13 @@
 (s/defn read-entities
   "Retrieve many entities of the same type provided their ids and common type"
   [ids entity-type auth-identity
-   services :- ReadEntitiesServices]
-  (let [read-entity (read-fn entity-type auth-identity {} services)]
-    ;; TODO search by ids
-    (pmap (fn [id]
-           (try
-             (some-> (read-entity id)
-                     (with-long-id services))
-             (catch Exception e
-               (log/error (pr-str e)))))
-         ids)))
+   {{:keys [get-store]} :StoreService
+    :as services} :- ReadEntitiesServices]
+  (let [store (get-store entity-type)]
+    (map #(when %
+            (with-long-id % services))
+         (binding [es-crud/*throw-access-control-error?* false]
+           (store/read-records store ids auth-identity {})))))
```

Comment:
the behaviour can't be the same because, in the original implementation, some records reading throws an error and try/catch replaces it with nil. And read-records returns entier collection of records. 

Also, it looks like to me that the original implementation was meant to wrap with try/catch the call to `with-long-id`. WDYT?

Reply:
Now it should behave the same. Unfortunately, there is a small code duplication ( wdyt?